src/laboratory/static/js/shelfobjectedit.js

```diff
@@ -4,11 +4,35 @@ function processResponseshelfobject(dat){
     $("#closemodal").html(dat["inner-fragments"]["#closemodal"]);
     datatableelement.ajax.reload();
 }
-
-function processResponseshelfobjectDelete(dat) {
-	$('#shelfobjectDelete').html(dat);
-	$("#object_delete").modal('show');
+//Refactored delete method for Shelf Object
+function shelfObjectDelete(obj, shelf_object_id, text) {
+    let url = $(obj).data('url')
+    Swal.fire({ //Confirmation for delete
+        title: 'Are you sure?',
```

Comment:
I think this confirmation message should include more information to help the user.  Like "Are you sure you want to delete '<whatever name the obj has>'?  This way the user can make sure they are deleting what they need.  In my experience it will help prevent errors.

Reply:
Added shelf object name to message